File: src/pkg/player/8durt/lib/Redstone.lua
-- package: lib

require("ktox-lib")
ktox_sourcemap_traceback(debug and debug.getinfo and (debug.getinfo(1) or {}).short_src or "", "lib/Redstone.kt", {["1-9"]=1,["10"]=12,["11"]=13,["12-13"]=14,["14"]=19,["15"]=20,["16"]=21,["17"]=22,["18-20"]=23}, "lib")

---@param jobType string
---@param on boolean
---@return boolean
function setJobPower(jobType, on)
    local relaySide = ktoxConfigRelayForJob(jobType)
    if relaySide == "MISSING" then
        return false
    end
    local parts = ktox_split(relaySide, ",")
    local relayName = parts[1]
    local side = parts[2]
    local result = ktoxPeripheralCall(relayName, "setOutput", "S:" .. tostring(side) .. "|B:" .. tostring(on))
    return result ~= "MISSING"
end

File: ktox-lua-cc/src/main/kotlin/lib/Redstone.kt
package lib

import common.ktoxConfigRelayForJob
import common.ktoxPeripheralCallRaw

// Toggles the redstone relay (see PLAN.md — a real CC:Tweaked peripheral
// mirroring the `redstone` global API, reachable remotely over the wired
// network) that controls the given job/machine type, per
// config/peripherals.json. Returns false if no relay is configured for
// this job type, or the relay peripheral isn't reachable.
fun setJobPower(jobType: String, on: Boolean): Boolean {
    val relaySide = ktoxConfigRelayForJob(jobType)
    if (relaySide == "MISSING") {
        return false
    }
    // ktox does NOT offset List indexing from Kotlin's 0-based convention
    // to Lua's 1-based tables - parts[1] is the first element. See
    // AGENTS.md and lib/Position.kt for the same idiom.
    val parts = relaySide.split(",")
    val relayName = parts[1]
    val side = parts[2]
    val result = ktoxPeripheralCallRaw(relayName, "setOutput", "S:${side}|B:${on}")
    return result != "MISSING"
}
